### PR TITLE
Add representation of the pt_graph in pt_ref's documentation

### PR DESCRIPTION
**Pull request by nonifier**
Add a representation of the graph that we have in pt_ref. 
This can be helpful to understand how the data is structured and how we resolve conversion from one type to another.

peek : https://github.com/nonifier/navitia/blob/pt_graph_docs/documentation/rfc/ptref_grammar.md#data-graph

### CODE REVIEW

**Review by pbougue — DISMISSED** [review]
Nice idea!
Is it possible to automatize that?
Another idea would be to provide source, add how-to regenerate, and comment the code part to ask for doc update.

**Comment by benoit-bst** [comment]
@pbougue, to undestand how to create a **Route**, and how it is attach to the others :ok_hand:

**Review by benoit-bst — APPROVED** [review]

**Comment by nonifier** [comment]
@pbougue : you commented too fast... I was adding a link to graphviz-online to easily generate the graph. For the auto-gen, I'll let you have a look at it ;)

**Comment by pbougue** [comment]
For later try, something along this seems to work:
```bash
grep "boost::add_edge(vertex_map.at(Type_e::\([^)]*\)), vertex_map.at(Type_e::\([^)]*\)), g);" source/ptreferential/ptref_graph.cpp | sed "s# *boost::add_edge(vertex_map.at(Type_e::\([^)]*\)), vertex_map.at(Type_e::\([^)]*\)), g);#\1 -> \2#g"
grep "boost::add_edge(vertex_map.at(Type_e::\([^)]*\)), vertex_map.at(Type_e::\([^)]*\)), Edge(\([^)]*\)), g);" source/ptreferential/ptref_graph.cpp | sed "s# *boost::add_edge(vertex_map.at(Type_e::\([^)]*\)), vertex_map.at(Type_e::\([^)]*\)), Edge(\([^)]*\)), g);#\1 -> \2 [ label=\"\3\" ]#g"
echo 'Impact -> Network  [ label="100" ]\nImpact -> Line  [ label="100" ]\nImpact -> Route  [ label="100" ]\nImpact -> StopArea  [ label="100" ]\nImpact -> StopPoint  [ label="100" ]\nImpact -> MetaVehicleJourney  [ label="100" ]\nImpact -> VehicleJourney  [ label="100" ]'
```
Where it would be safer to compare the result to `grep 'boost::add_edge(' source/ptreferential/ptref_graph.cpp` :wink: 

Anyway, looks like you missed `Impact -> VehicleJourney  [ label="100" ]`
And the graph needs to be regenerated anyway, there is a `)` in the weight of Network->Dataset in image.

@benoit-bst it may help, yes :sweat_smile:

**Comment by TeXitoi** [comment]
https://www.boost.org/doc/libs/1_69_0/libs/graph/doc/write-graphviz.html

File: documentation/rfc/ptref_grammar.md [hunk 1]
```diff
@@ -165,6 +165,15 @@ Examples:
 * `/v1/coverage/id/lines?filter=get network <- line.id=Metro14` returns the lines from the network of the line Metro14.
 * `/v1/coverage/id/physical_modes?filter=get connection <- line.id=Metro14` returns the physical modes in connection with the line Metro14.
 
+## Data Graph
+The data is structured as a graph, where related objects are connected. When converting a collection into another, we use the graph to define the shortest series of jointures that needs to be performed to get to the target collection.
+
+The graph is as follow (edges may have weight to avoid unwanted behavior):
```
Comment (TeXitoi) [inline documentation/rfc/ptref_grammar.md:171]: As this is a functional doc, all shortcuts must be removed from the graph. The user can't known how the shortcuts are generated.

Comment (pbougue) [reply documentation/rfc/ptref_grammar.md:171]: I don't know if it should be removed from functional doc (it's already a pretty technical need).
But shortcuts need to be documented for technical purpose I think (debug and maintenance).

File: documentation/rfc/ptref_grammar.md [hunk 2]
```diff
@@ -172,7 +172,7 @@ The graph is as follow (edges may have weight to avoid unwanted behavior):
...
...A%20%20%20%20%20%20%20%20%20%20%3Ctr%3E%3Ctd%20align%3D%22right%22%20port%3D%22i3%22%3E%3Cfont%20color%3D%22green4%22%3EImpact%20link%3C%2Ffont%3E%3C%2Ftd%3E%3C%2Ftr%3E%0A%20%20%20%20%20%20%20%20%20%20%3C%2Ftable%3E%3E%5D%0A%20%20%20%20%20%20%20%20key2%20%5Blabel%3D%3C%3Ctable%20border%3D%220%22%20cellpadding%3D%222%22%20cellspacing%3D%220%22%20cellborder%3D%220%22%3E%0A%20%20%20%20%20%20%20%20%20%20%3Ctr%3E%3Ctd%20port%3D%22i1%22%3E%26nbsp%3B%3C%2Ftd%3E%3C%2Ftr%3E%0A%20%20%20%20%20%20%20%20%20%20%3Ctr%3E%3Ctd%20port%3D%22i2%22%3E%26nbsp%3B%3C%2Ftd%3E%3C%2Ftr%3E%0A%20%20%20%20%20%20%20%20%20%20%3Ctr%3E%3Ctd%20port%3D%22i3%22%3E%26nbsp%3B%3C%2Ftd%3E%3C%2Ftr%3E%0A%20%20%20%20%20%20%20%20%20%20%3C%2Ftable%3E%3E%5D%0A%20%20%20%20%20%20%20%20key%3Ai1%3Ae%20-%3E%20key2%3Ai1%3Aw%20%5B%20label%3D%22weight%3D1%22%20style%3D%22bold%22%5D%0A%20%20%20%20%20%20%20%20key%3Ai2%3Ae%20-%3E%20key2%3Ai2%3Aw%20%5B%20label%3D%22weight%3Dx%22%20fontcolor%3D%22red%22%20color%3D%22red%22%20%5D%0A%20%20%20%20%20%20%20%20key%3Ai3%3Ae%20-%3E%20key2%3Ai3%3Aw%20%5B%20label%3D%22weight%3D100%22%20style%3D%22dashed%22%20fontcolor%3D%22green4%22%20color%3D%22green4%22%20%5D%0A%20%20%20%20%20%20%7D%0A%20%20%0A%7D)
```
Comment (pbougue) [inline documentation/rfc/ptref_grammar.md:175]: minor: '→ Edit graph' (no `_`)

**Review by pbougue — APPROVED** [review]

**Pull request merged**